src/Nowruz/modules/userProfile/containers/createUpdateCertificate/useCreateUpdateCertificate.tsx

```diff
@@ -0,0 +1,262 @@
+import { useDispatch, useSelector } from 'react-redux';
+import { createOrganization, getOrganization, Organization, otherProfileByUsername, search, User } from 'src/core/api';
+import { createAdditional, removeAdditional, updateAdditional } from 'src/core/api/additionals/additionals.api';
+import { AdditionalReq, AdditionalRes, CertificateMeta } from 'src/core/api/additionals/additionals.types';
+import { monthNames } from 'src/core/time';
+import { removedEmptyProps } from 'src/core/utils';
+import { Avatar } from 'src/Nowruz/modules/general/components/avatar/avatar';
+import { RootState } from 'src/store';
+import { setUser } from 'src/store/reducers/profile.reducer';
+import * as yup from 'yup';
+
+const schema = yup
+  .object()
+  .shape({
+    name: yup.string().required('Required'),
+    orgName: yup.string().required('Required'),
+    orgId: yup.string(),
+    issueMonth: yup.string(),
+    issueYear: yup.string(),
+    expireMonth: yup.string(),
+    expireYear: yup.string(),
+    credentialId: yup.string(),
+    credentialUrl: yup
+      .string()
+      .matches(
+        /^$|((https?):\/\/)?(www.)?[a-z0-9]+(\.[a-z]{2,}){1,3}(#?\/?[a-zA-Z0-9#]+)*\/?(\?[a-zA-Z0-9-_]+=[a-zA-Z0-9-%]+&?)?$/,
+        'Enter correct url!',
+      ),
+    description: yup.string(),
+  })
+  .required();
+
+interface OptionType {
+  value: string;
+  label: string;
+}
+
+export const useCreateUpdateCertificate = (
+  handleClose: () => void,
+  certificate: AdditionalRes,
+  setCertificate: (val: AdditionalRes) => void,
+) => {
+  const user = useSelector<RootState, User | undefined>((state) => {
+    return state.profile.user;
+  });
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+    getValues,
+    setValue,
+  } = useForm({
+    mode: 'all',
+    resolver: yupResolver(schema),
+  });
+
+  const dispatch = useDispatch();
+
+  const [orgVal, setOrgVal] = useState<OptionType>();
+  const [orgs, setOrgs] = useState<Organization[]>([]);
+  const [months, setMonths] = useState<OptionType[]>([]);
+  const [years, setYears] = useState<OptionType[]>([]);
+  const [issueMonth, setIssueMonth] = useState<OptionType>();
+  const [issueYear, setIssueYear] = useState<OptionType>();
+  const [expMonth, setExpMonth] = useState<OptionType>();
+  const [expYear, setExpYear] = useState<OptionType>();
+
+  const mapMonthNames = () => {
+    const options = monthNames.map((m, index) => {
+      return { value: index.toString(), label: m };
+    });
+    setMonths(options);
+  };
+
+  const getYearOptions = () => {
+    const currentYear = new Date().getFullYear();
+    const start = currentYear - 40;
+    const options: OptionType[] = [];
+    for (let i = start; i <= currentYear; i++) {
+      const year = i.toString();
+      options.push({ value: year, label: year });
+    }
+    setYears(options);
+  };
+
+  const initializeValues = () => {
+    const meta = certificate ? (certificate.meta as CertificateMeta) : null;
```

Comment:
Suggest use react hook form defaultValue for init

Reply:
Default value is not applicable here, since default values are empty(for creating new entity) and I call initialize function in case of updating an existing entity and I want to initialize the form with that entity. However, thanks to you, I found that reset function of reactHookForm is a better option in this case